lib/extensions.js

```diff
@@ -379,6 +403,45 @@ module.exports = function(realmConstructor) {
                     return classPermissions[0];
                 }
             },
+
+            subscriptions(arg) {
+                let all_subscriptions = this.objects('__ResultSets');
+                if (arg) {
+                    if (arg.includes('*') || arg.includes('?')) {
+                        all_subscriptions = all_subscriptions.filtered(`name LIKE '${arg}'`);
+                    } else {
+                        all_subscriptions = all_subscriptions.filtered(`name == '${arg}'`);
+                    }
+                }
+                let listOfSubscriptions = [];
+                for (var i = 0; i < all_subscriptions.length; ++i) {
+                    let matches_property = all_subscriptions[i]['matches_property'];
+                    let sub = {
+                        name: all_subscriptions[i]['name'],
+                        objectType: matches_property.substr(0, matches_property.length-8), // remove _matches
+                        query: all_subscriptions[i]['query'],
+                    }
+                    listOfSubscriptions.push(sub);
+                }
+                return listOfSubscriptions;
+            },
+
+            unsubscribe(name) {
+                if (name && typeof(name) === 'string') {
+                    let named_subscriptions = this.objects('__ResultSets').filtered(`name == '${name}'`);
```

Comment:
A minor optimization would be to return early if there are no matches.

Reply:
Avoiding an empty transaction good practice.